### PR TITLE
[BEAM-3388] Avoid various reflection and re-processing in the Go SDK runtime

### PR DESCRIPTION
**Pull request by herohde**

### CODE REVIEW

**Comment by herohde** [comment]
R: @wcn3

**Comment by herohde** [comment]
Any comments or concerns?

File: sdks/go/pkg/beam/core/runtime/exec/input.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	defer inputsMu.Unlock()
+
+	key := t.String()
+	if _, exists := inputs[key]; exists {
+		log.Warnf(context.Background(), "Input for %v already registered. Overwriting.", key)
+	}
+	inputs[key] = maker
+}
+
+type reIterValue struct {
```
Comment (wcn3) [inline sdks/go/pkg/beam/core/runtime/exec/input.go]: This casing seems incorrect. Camel-casing is general used to mark word boundaries, but here, it's emphasizing the prefix 're'. I think this should be 'reiterValue'

Comment (herohde) [reply sdks/go/pkg/beam/core/runtime/exec/input.go]: The name is also meant to match the "ReIter" constant. If both should be "Reiter" then we should probably come up with a better name altogether. I prefer to address such a change in a separate PR.

**Comment by herohde** [comment]
Thanks!

**Comment by herohde** [comment]
R: @robertwb (merge)

File: sdks/go/pkg/beam/core/runtime/exec/input.go (hunk 1 above)
Comment (wcn3) [reply sdks/go/pkg/beam/core/runtime/exec/input.go]: I don't have a great suggestion here. I agree this comment is not a blocker for this PR.

**Pull request merged**